include/libdnf/rpm/transaction.hpp

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_RPM_TRANSACTION_HPP
+#define LIBDNF_RPM_TRANSACTION_HPP
+
+#include "libdnf/utils/exception.hpp"
+#include "libdnf/utils/weak_ptr.hpp"
+
+#include <memory>
+
+namespace libdnf {
+
+class Base;
+
+}  // namespace libdnf
```

Comment:
Out of curiosity, why don't you do:
```c++
namespace libdnf {

class Base;

namespace rpm {

namespace solv {
class SolvPrivate;
} // solv

// all the code

}} // libdnf::rpm
```

Reply:
I'd keep it as it is, I find it better than the old-style nesting.